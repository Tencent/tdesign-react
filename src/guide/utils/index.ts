import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
