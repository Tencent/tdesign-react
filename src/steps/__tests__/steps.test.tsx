import React from 'react';
import { render, waitFor, fireEvent, vi } from '@test/utils';
import Steps from '../Steps';

const { StepItem } = Steps;

const stepOptions = [
  {
    title: '1',
    content: '这里是提示文字',
    value: 1,
  },
  {
    title: '2',
    content: '这里是提示文字',
    value: 2,
  },
  {
    title: '3',
    content: '这里是提示文字',
    value: 3,
  },
];

describe('Steps 组件测试', () => {
  test('mount, unmount 测试', () => {
    const wrapper = render(
      <Steps current={1}>
        <StepItem status="finish" title="1" content="这里是提示文字"></StepItem>
        <StepItem status="process" title="2" content="这里是提示文字"></StepItem>
        <StepItem status="error" title="3" content="这里是提示文字"></StepItem>
        <StepItem title="4" content="这里是提示文字"></StepItem>
      </Steps>,
    );

    expect(() => {
      wrapper.unmount();
    }).not.toThrow();
  });

  test('options 测试', async () => {
    const testId = 'step options test';
    const handleChange = vi.fn();

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Steps current={2} options={stepOptions} onChange={handleChange} />
      </div>,
    );

    const stepsInstance = await waitFor(() => getByTestId(testId));
    const stepsItems = stepsInstance.querySelectorAll('.t-steps-item');
    expect(stepsItems.length).toBe(3);

    fireEvent.click(stepsInstance.querySelector('.t-steps-item__inner'));
    expect(handleChange).toBeCalledTimes(1);
  });

  test('layout vertical 测试', async () => {
    const testId = 'step layout test';

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Steps layout="vertical" options={stepOptions} />
      </div>,
    );

    const stepsInstance = await waitFor(() => getByTestId(testId));
    const stepsItems = stepsInstance.querySelectorAll('.t-steps--vertical');
    expect(stepsItems.length).toBe(1);
  });

  test('layout readonly 测试', async () => {
    const testId = 'step readonly test';

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Steps options={stepOptions} readonly />
      </div>,
    );

    const stepsInstance = await waitFor(() => getByTestId(testId));
    const stepsItems = stepsInstance.querySelectorAll('.t-steps-item--clickable');
    expect(stepsItems.length).toBe(0);
  });
});
