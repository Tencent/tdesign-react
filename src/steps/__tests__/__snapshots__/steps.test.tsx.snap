// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extra.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--default-anchor t-steps--vertical t-steps--positive t-steps--line-separator"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-finish"
        >
          <span
            class="t-steps-item__icon--number"
          >
            <svg
              class="t-icon t-icon-check"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            步骤1
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item__icon t-steps-item-process"
        >
          <span
            class="t-steps-item__icon--number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            步骤2
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
          <div
            class="t-steps-item__extra"
          >
            <button
              class="t-button t-button--theme-default t-button--variant-text t-size-s"
              type="button"
            >
              <span
                class="t-button__text"
              >
                pre
              </span>
            </button>
            <div
              style="display: inline; margin-left: 4px;"
            >
              <button
                class="t-button t-button--theme-primary t-button--variant-base t-size-s"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  Next
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-default"
        >
          <span
            class="t-steps-item__icon--number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            步骤3
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-default"
        >
          <span
            class="t-steps-item__icon--number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            步骤4
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--default-anchor t-steps--horizontal t-steps--positive t-steps--line-separator"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-finish"
        >
          <span
            class="t-steps-item__icon--number"
          >
            <svg
              class="t-icon t-icon-login"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8.48 7.5L6.23 5.25l.7-.7 3.1 3.1c.2.2.2.5 0 .7l-3.1 3.1-.7-.7L8.48 8.5H1v-1h7.48z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M4 5V3h8v10H4v-2H3v2.5c0 .28.22.5.5.5h9a.5.5 0 00.5-.5v-11a.5.5 0 00-.5-.5h-9a.5.5 0 00-.5.5V5h1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            登录
          </div>
          <div
            class="t-steps-item__description"
          >
            已完成状态
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item__icon t-steps-item-process"
        >
          <span
            class="t-steps-item__icon--number"
          >
            <svg
              class="t-icon t-icon-cart"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M1 3h1.58l1.39 8.33c.06.39.4.67.78.67H14v-1H4.92L4.6 9h8.63a.8.8 0 00.78-.6l.85-3.4a.8.8 0 00-.78-1H3.76l-.23-1.33A.8.8 0 002.75 2H1v1zm12.07 5H4.42l-.5-3h9.9l-.75 3zM4.75 14.5a.75.75 0 100-1.5.75.75 0 000 1.5zM12.81 14.5a.75.75 0 100-1.5.75.75 0 000 1.5z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            购物
          </div>
          <div
            class="t-steps-item__description"
          >
            进行中状态
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-default"
        >
          <span
            class="t-steps-item__icon--number"
          >
            <svg
              class="t-icon t-icon-wallet"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M11.5 4.5h2a1 1 0 011 1V12a1 1 0 01-1 1h-11a1 1 0 01-1-1V3a1 1 0 011-1h8a1 1 0 011 1v1.5zm-1-1.5h-8v1.5h8V3zm3 2.5h-11V12h11V5.5z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            支付
          </div>
          <div
            class="t-steps-item__description"
          >
            未开始
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-default"
        >
          <span
            class="t-steps-item__icon--number"
          >
            <svg
              class="t-icon t-icon-check-circle"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M4.5 8.2L7 10.7l4.5-4.5-.7-.7L7 9.3 5.2 7.5l-.7.7z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M4.11 2.18a7 7 0 117.78 11.64A7 7 0 014.1 2.18zm.56 10.8a6 6 0 106.66-9.97A6 6 0 004.67 13z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            完成
          </div>
          <div
            class="t-steps-item__description"
          >
            未开始
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`no-sequence.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--dot-anchor t-steps--horizontal t-steps--positive t-steps--line-separator"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-finish"
        />
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item__icon t-steps-item-process"
        />
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-default"
        />
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner t-steps-item--clickable"
      >
        <div
          class="t-steps-item__icon t-steps-item-default"
        />
        <div
          class="t-steps-item__content"
        >
          <div
            class="t-steps-item__title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item__description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`sequence.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-steps t-steps--default-anchor t-steps--horizontal t-steps--positive t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          >
            <span
              class="t-steps-item__icon--number"
            >
              2
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          >
            <span
              class="t-steps-item__icon--number"
            >
              3
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          >
            <span
              class="t-steps-item__icon--number"
            >
              4
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps t-steps--default-anchor t-steps--horizontal t-steps--positive t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          >
            <span
              class="t-steps-item__icon--number"
            >
              3
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          >
            <span
              class="t-steps-item__icon--number"
            >
              4
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`status.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="t-steps t-steps--default-anchor t-steps--horizontal t-steps--positive t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          >
            <span
              class="t-steps-item__icon--number"
            >
              2
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-wait"
          >
            <span
              class="t-steps-item__icon--number"
            >
              3
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-wait"
          >
            <span
              class="t-steps-item__icon--number"
            >
              4
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps t-steps--default-anchor t-steps--horizontal t-steps--positive t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          >
            <span
              class="t-steps-item__icon--number"
            >
              2
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--error"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-error"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-close"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              错误的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              优先展示\`t-step\`中设置的 status
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-wait"
          >
            <span
              class="t-steps-item__icon--number"
            >
              4
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`vertical-no-sequence.jsx 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <div
      class="t-steps t-steps--dot-anchor t-steps--vertical t-steps--positive t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps t-steps--dot-anchor t-steps--vertical t-steps--reverse t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          />
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`vertical-sequence.jsx 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <div
      class="t-steps t-steps--default-anchor t-steps--vertical t-steps--positive t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          >
            <span
              class="t-steps-item__icon--number"
            >
              2
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          >
            <span
              class="t-steps-item__icon--number"
            >
              3
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          >
            <span
              class="t-steps-item__icon--number"
            >
              4
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps t-steps--default-anchor t-steps--vertical t-steps--reverse t-steps--line-separator"
    >
      <div
        class="t-steps-item t-steps-item--wait"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-default"
          >
            <span
              class="t-steps-item__icon--number"
            >
              4
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--process"
      >
        <div
          class="t-steps-item__inner"
        >
          <div
            class="t-steps-item__icon t-steps-item-process"
          >
            <span
              class="t-steps-item__icon--number"
            >
              3
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              未进行的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              进行中的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-steps-item t-steps-item--finish"
      >
        <div
          class="t-steps-item__inner t-steps-item--clickable"
        >
          <div
            class="t-steps-item__icon t-steps-item-finish"
          >
            <span
              class="t-steps-item__icon--number"
            >
              <svg
                class="t-icon t-icon-check"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
          <div
            class="t-steps-item__content"
          >
            <div
              class="t-steps-item__title"
            >
              已完成的步骤
            </div>
            <div
              class="t-steps-item__description"
            >
              这里是提示文字
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
